### PR TITLE
ENH: Deprecate val_set_is_dist 

### PR DESCRIPTION
**Pull request by kenza-bouzid**
-No need to check if validation set is distributed now that we can shard evaluation (test and val) dataloaders without duplications #743 (it will be distributed anyways)
-Tests tiles_selction, outputs_gathering and loss_callback are updated to account for uneven samples scenario

### CODE REVIEW

**Review by ant0nsc — APPROVED** [review]

**Comment by harshita-s** [comment]
Some changes were made in the tests related to uneven samples, could you please explain in the PR description

**Review by harshita-s — APPROVED** [review]

File: hi-ml-cpath/testhisto/testhisto/utils/test_callbacks.py [hunk 1]
```diff
@@ -25,12 +25,19 @@ def _assert_loss_cache_contains_n_elements(loss_cache: LossCacheDictType, n: int
         assert len(loss_cache[key]) == n
 
 
-def get_loss_cache(n_slides: int = 4, rank: int = 0) -> LossCacheDictType:
+def get_loss_cache(n_slides: int = 4, offset: int = 0, rank: int = 0) -> LossCacheDictType:
+    """Get a loss cache with n_slides elements.
+
+    :param n_slides: The number of slides, defaults to 4
+    :param offset: An offset to simulate uneven samples scenario, defaults to 0
```
[the remark this answers is not in the document]
Comment (kenza-bouzid) [reply hi-ml-cpath/testhisto/testhisto/utils/test_callbacks.py:32]: I added an offset to simulate uneven samples. The loss_callback is able to detect duplicates and eliminates them, so we need to add an offset to SLIDE_ID to make sure the mock data won't have duplicates and at the same time we reduce the number of samples of rank 1 by 1 sample as if we're using the UnrepeatedDistributedSampler #743 
this is important to make sure all gather_object operations are still working properly even if there is uneven samples
